Python/compile.c

```diff
@@ -4976,16 +4944,79 @@ stackdepth_walk(struct compiler *c, basicblock *b, int depth, int maxdepth)
 static int
 stackdepth(struct compiler *c)
 {
-    basicblock *b, *entryblock;
-    entryblock = NULL;
+    basicblock *b, *entryblock = NULL;
+    basicblock **stack, **sp;
+    int nblocks = 0, depth = 0, maxdepth = 0;
     for (b = c->u->u_blocks; b != NULL; b = b->b_list) {
         b->b_seen = 0;
         b->b_startdepth = INT_MIN;
         entryblock = b;
+        nblocks++;
     }
     if (!entryblock)
         return 0;
-    return stackdepth_walk(c, entryblock, 0, 0);
+    stack = (basicblock **)PyObject_Malloc(sizeof(basicblock *) * nblocks);
+    if (!stack) {
+        PyErr_NoMemory();
+        return -1;
+    }
+
+    sp = stack;
+    stackdepth_push(&sp, entryblock, 0);
+    while (sp != stack) {
+        int i;
+        basicblock *next;
+        b = sp[-1];
+        if (b->b_seen == 2) {
```

Comment:
Perhaps add some comment as to the different values of `b_seen`? Does 1 mean "queued for processing" and 2 "already processed"?